### ISSUE TITLE
Website: issues with dialog panel slide transition in Safari

### ISSUE THREAD

**Issue by ianmcburnie** [eBay/skin#434]
On the production website both of the DS6 left/right dialog panel transitions work without issue, but neither of the DS4 transitions work anymore (this regression crept in during the v6 release).

In the `6.1.0-prerebase` branch both the DS4 and DS6 left panel transitions work, but the right panel transition is broken in both. At least it's now consistent (they are both using the same JS after my recent clear out)!!!

<img width="301" alt="screen shot 2018-10-23 at 10 10 39 am" src="https://user-images.githubusercontent.com/38065/47378471-13857600-d6ad-11e8-9b74-ddff0b94969b.png">

**Comment by ianmcburnie** [comment]
Found the culprit. This line:

https://github.com/eBay/skin/pull/419/files#diff-aad745914bc925444964eacb0ccc0a04R120

Odd, that setting focus on the close button should do this.

**Comment by seangates** [comment]
Okay, that's weird. Did anything change in the `display` value of the parent, or the element, that would cause Skin to re-render this on the wrong side?

**Comment by ianmcburnie** [comment]
Just verified that the same issue occurs in master branch when I add that line. Setting a timeout might help. We should check how we set focus in eBayUI and whether this could be causing any of the recent issues in the dialog component.

**Comment by DylanPiercey** [comment]
I know safari forces focused elements to become visible basically no matter what. In ebayui-core I got around this by waiting for the transition to finish before applying the new focus. Could that be the issue here?

**Comment by ianmcburnie** [comment]
Thanks @DylanPiercey . That lead me to this snippet of code in the website, which I didn't notice before:

```js
function handleTransitionEnd (isOpening) {
        // focus on the close button
        if (isOpening) {
            dialogClose.focus();
        }

        cancel = undefined;
    }
```

But focus doesn't seem to be set on the button unless I put it behind an additional timeout: 

```js
function handleTransitionEnd (isOpening) {
        // focus on the close button
        if (isOpening) {
            window.setTimeout(function() {
                dialogClose.focus();
            }, 100);
        }

        cancel = undefined;
    }
```

And when a shorter timeout is used, it exhibits the problems mentioned in the description above.

**Comment by ianmcburnie** [comment]
I see the `transition.js` util in Skin is a little out of sync with that in eBayUI. Let me try upgrading the one in Skin and see if it resolves the `handleTransitionEnd` issue.

**Comment by ianmcburnie** [comment]
Upgrading to latest transition util created more issues than it solved. I'm just going to put in the quick fix above for this for now until we get a chance to investigate the transition utility further.

**Comment by ianmcburnie** [comment]
Fixed in 6.1.0 branch.
